build_config/host-shared.rb

```diff
@@ -29,6 +26,9 @@
   # file separator
   # conf.file_separator = '/'
 
+  # for better compatibility with C++
+  conf.enable_cxx_exception
```

Comment:
I am concerned about the reliability of `enable_cxx_exception` (especially performance). Is it ok to turn it on all the time?

Reply:
I don't know for sure about reliability, so I commented out.
As for performance, I would like to find a way to improve it, since the degradation due to shared objectization is more noticeable.

The benchmark results are attached.
Settings can be found <https://gist.github.com/dearblue/8f4e58dd34b5c10a40e6bc605c32bf79#file-shared-benchmark>.
The run command is `ruby run-benchmark.rb shared.benchmark`.

![bm-result-20231025T213440](https://github.com/mruby/mruby/assets/6077921/c46d0f1e-b0c4-4abe-9e4c-a3b536c1b260)